jujugui/static/gui/src/app/app.js

```diff
@@ -1892,9 +1892,14 @@ YUI.add('juju-gui', function(Y) {
       @param {String} uuid The environment UUID where to switch to.
       @param {String} username The username for the new environment.
       @param {String} password The password for the new environment.
+      @param {Boolean} reconnect Whether to reconnect to a new environment.
     */
-    switchEnv: function(socketUrl, username, password) {
-      console.log('switching to new socket URL:', socketUrl);
+    switchEnv: function(socketUrl, username, password, reconnect=!!socketUrl) {
+      if (reconnect) {
+        console.log('switching to new socket URL:', socketUrl);
```

Comment:
Do we need these logs? Are they more of a debugging thing? If it's a user facing thing can we put it in the UI as a notification?